scripts/validate-openapi/run.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# exit when any command fails
+set -e
+
+# Check if docker is installed
+if ! command -v "docker" &> /dev/null
+then
+    echo "Unable to find docker. Is it installed and added to your \$PATH?"
+    exit 1
+fi
+
+DOCKER_REGISTRY="567716553783.dkr.ecr.us-east-1.amazonaws.com"
+DOCKER_IMAGE="engineering-docs"
+DOCKER_IMAGE_TAG="builder"
+
+# Check if user is logged in to DOCKER_REGISTRY
+docker login -u=${DOCKERUSERNAME} -p=${DOCKERPASSWORD} ${DOCKER_REGISTRY}
+docker pull ${DOCKER_REGISTRY}/${DOCKER_IMAGE}:${DOCKER_IMAGE_TAG}
+exit_code=$?
+
+if [ ! $exit_code -eq 0 ]; then
+    echo "Unable to pull ${DOCKER_IMAGE}:${DOCKER_IMAGE_TAG} image. Are you logged in ${DOCKER_REGISTRY}?"
+    exit 1
+fi
+
+if [ ! -f $(pwd)/openapi.yaml ]; then
```

Comment:
mmm this will execute the check regardless if the file has been modified or not in the PR. I´d challenge this solution to be executed ONLY if the file has been modified. 

Maybe we could implement a condition (following this example https://github.com/Typeform/ci-standard-checks/blob/main/src/conditions/triggeredByBot.ts#L15) that checks in the context if openapi.yaml is in the list of changed files.

@kaofelix you may advise here better than me

Reply:
This is a good point, I forgot to add it in the TODO list :joy:

Reply:
Hey @erusso7! Here's an idea: in the secret scan script, we do a call to the github API to list commits in a PR: https://github.com/Typeform/ci-standard-checks/blob/main/scripts/secrets-scan/run.sh#L51-L59

Instead, you could use similar code to call the PR files endpoint, so you could check the resulting list to see if your file is included there or not: https://docs.github.com/en/rest/reference/pulls#list-pull-requests-files. There's probably a way to do that with only some `git` CLI magic, but I find a little fiddly and some of the obvious solutions (like checking `main..<pr_head_ref>`) can fail in some not so exceptional cases